We know that the kernel headers are in place. Don't try to install them.

diff -dur uClibc-0.9.28.1.orig/Makefile uClibc-0.9.28.1/Makefile
--- uClibc-0.9.28.1.orig/Makefile	2007-01-28 06:38:00.000000000 +0100
+++ uClibc-0.9.28.1/Makefile	2007-01-28 18:55:46.000000000 +0100
@@ -158,12 +158,10 @@
 	$(INSTALL) -d $(PREFIX)$(DEVEL_PREFIX)lib
 	$(INSTALL) -d $(PREFIX)$(DEVEL_PREFIX)include
 	-$(INSTALL) -m 644 lib/*.[ao] $(PREFIX)$(DEVEL_PREFIX)lib/
-	if [ "$(KERNEL_SOURCE)" = "$(DEVEL_PREFIX)" ] ; then \
-		extra_exclude="--exclude include/linux --exclude include/asm'*'" ; \
-	else \
-		extra_exclude="" ; \
-	fi ; \
-	tar -chf - include --exclude .svn --exclude CVS $$extra_exclude \
+	extra_exclude="--exclude include/linux --exclude include/asm'*'" ; \
+	tar -chf - include --exclude .svn --exclude CVS \
+	                   --exclude include/linux      \
+	                   --exclude include/asm'*'     \
 		| tar -xf - -C $(PREFIX)$(DEVEL_PREFIX)
 	echo '/* Dont use _syscall#() macros; use the syscall() function */' > \
 		$(PREFIX)$(DEVEL_PREFIX)include/bits/syscalls.h
